models/users.ts

```diff
@@ -0,0 +1,13 @@
+import { object, string, TypeOf } from "yup";
+
+export const userSchema = object({
+  id: string().ensure().required(),
+  first_name: string().ensure().required(),
+  last_name: string().ensure().required(),
+  email: string().ensure().required(),
+  role: string().ensure().required().oneOf(["Admin", "Volunteer", "Teacher", "Student", "Parent"]),
+  phone_number: string().optional(),
+  address: string().ensure().required(),
```

Comment:
Alright. Now that we have transitioned to `InferType`, here is how yup works:

- `ensure()` will typecast nulls and undefined into empty strings
- `defined()` means that all values except undefined are ok.
- `required()` means all non null value. 

So if we want to define a value in our model that is a string and nullable, it would be like `id: string().nullable().defined()`. 

So, using that, we can remove the `ensure()'s` here. None of these values should be null for right now, so keep them as `required()`,

However, this means that we need to have a second class called updateUser, which has all of these values except for id and all of them are optional. So create that second class here and use it instead of `User` in patch requests

Reply:
@anvitaa22